db/disk/course.go

```diff
@@ -12,10 +13,13 @@ import (
 const DISK_DB_COURSES_DIR = "courses";
 
 func (this *backend) ClearCourse(course *model.Course) error {
-    this.lock.Lock();
-    defer this.lock.Unlock();
+    courseDir := this.getCourseDir(course);
+    courseDir = util.ShouldAbs(courseDir);
```

Comment:
Instead of calling util.ShouldAbs() multiple times, just call it in getCourseDirFromID().

Reply:
Do you mean calling it from getCoursePathFromID(), since you said you I should lock the file over the directory?

Reply:
Take a look at the four path-creation functions at the bottom of this file.
I believe that all of them wrap getCoursePathFromID().
So if you abs there, they should all be absolute.

Reply:
Yes but it looks like getCoursePathFromID adds the config filename while getCourseDirFromID just returns the directory of the course. Ex. /autograder/courses/CS101 vs /autograder/courses/CS101/courseConfig.json. Since you mentioned locking the file (assuming you were mentioning the config file) instead of the directory, would it not make more sense to use getCoursePathFromID for the key?

Reply:
I think there are two key things here.
1) You should add abs to getCourseDirFromID(). I said the wrong function name before, but you probably saw that this is actually the function that is being wrapped.
2) It's important to conceptually think about what the locking key is and not just think of it as the output of a function. So I don't like this part from the previous comment: "would it not make more sense to use getCoursePathFromID for the key". We want to use the abs path to the course config file (which is held in the db). That can come from getCoursePathFromID(), but in other cases (like LoadCourse()), it is passed in directly (but needs to be normalized).